source/game/system/map/MapdataCheckPath.cc

```diff
@@ -21,10 +21,76 @@ void MapdataCheckPath::read(EGG::Stream &stream) {
     }
 }
 
+/// @brief Performs DFS to calculate @ref m_depth for all subsequent checkpaths.
+/// @param depth Number of checkpaths from first checkpath
+/// @addr{0x805150E0}
+void MapdataCheckPath::findDepth(s8 depth, const MapdataCheckPathAccessor &accessor) {
+    if (m_depth != -1) {
+        return;
+    }
+
+    m_depth = depth;
+
+    for (u16 nextID : m_next) {
+        if (nextID == 0xff) {
+            continue;
+        }
+
+        MapdataCheckPath *next = accessor.get(nextID);
+        next->findDepth(depth + 1, accessor);
+    }
+}
+
+u8 MapdataCheckPath::start() const {
+    return m_start;
+}
+
+u8 MapdataCheckPath::end() const {
+    return m_start + m_size - 1;
+}
+
+const std::array<u8, 6> &MapdataCheckPath::next() const {
+    return m_next;
+}
+
+const std::array<u8, 6> &MapdataCheckPath::prev() const {
+    return m_prev;
+}
+
+u8 MapdataCheckPath::depth() const {
+    return m_depth;
+}
+
+bool MapdataCheckPath::isPointInPath(u16 checkpointId) const {
+    return m_start <= checkpointId && checkpointId <= end();
+}
+
+/// @addr{0x80515014}
+MapdataCheckPath *MapdataCheckPathAccessor::findCheckpathForCheckpoint(u16 checkpointId) {
+    for (size_t i = 0; i < size(); i++) {
+        MapdataCheckPath *checkpath = get(i);
+        if (checkpath->isPointInPath(checkpointId)) {
+            return checkpath;
+        }
+    }
+    return nullptr;
+}
+
 MapdataCheckPathAccessor::MapdataCheckPathAccessor(const MapSectionHeader *header)
     : MapdataAccessorBase<MapdataCheckPath, MapdataCheckPath::SData>(header) {
     init(reinterpret_cast<const MapdataCheckPath::SData *>(m_sectionHeader + 1),
             parse<u16>(m_sectionHeader->count));
+
+    // maximum number of checkpaths one could traverse through in a lap
+    s8 maxDepth = -1;
+    get(0)->findDepth(0, *this);
+
+    for (size_t i = 0; i < size(); i++) {
+        s8 depth = get(i)->depth();
+        maxDepth = std::max(maxDepth, depth);
```

Comment:
As long as clang-format fits this in a single line, I think it is preferable to do this
```suggestion
        maxDepth = std::max(maxDepth, get(i)->depth());
```